src/net_processing.cpp

```diff
@@ -203,14 +203,32 @@ struct CNodeState {
      * otherwise: whether this peer sends non-witnesses in cmpctblocks/blocktxns.
      */
     bool fSupportsDesiredCmpctVersion;
-    //! Whether this peer is protected from disconnection due to a bad/slow chain
-    bool m_protect_from_disconnect;
-    //! A timeout used for checking whether our peer has sufficiently synced
-    int64_t m_headers_chain_timeout;
-    //! A header with the work we require on our peer's chain
-    const CBlockIndex * m_header_with_required_work;
-    //! After timeout is reached, set to true after sending getheaders
-    bool m_sent_getheaders_to_check_chain_sync;
+
+    /** State used to enforce CHAIN_SYNC_TIMEOUT
+      * Only in effect for outbound, non-manual connections, with
+      * m_protect == false
+      * Algorithm: if a peer's best known block has less work than our tip,
```

Comment:
Could maybe be more specific about the timeouts in the description. Instead of "T in the future", could write "CHAIN_SYNC_TIMEOUT seconds in the future", instead of "bump the timeout" could write "reset the timeout", instead of "new shorter timeout" could refer to HEADERS_RESPONSE_TIME.

Reply:
Done